packages/webapp/src/components/PolicyAndCookieBanner/PolicyAndCookieBanner.hooks.ts

```diff
@@ -0,0 +1,115 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { useEffect, useState } from 'react'
+
+import type {
+	CookieConsent,
+	CookieConsentBannerThemes,
+	CookieConsentManager,
+	WcpConsent as WC,
+} from './PolicyAndCookieBanner.types.js'
+
+declare global {
+	interface Window {
+		WcpConsent: WC | undefined
+	}
+}
+
+const COOKIE_CONSENT_BANNER_ID = `cookie-consent-banner-${
+	(Math.random() * 1000) | 0
```

Comment:
I assume this needs to change once the page reloads, is my assumption correct? If not, perhaps we could use `uniqueId` from lodash.

Reply:
This doesn't need to change at all and we probably don't need this. An earlier version of the component was a context provider allowing you to add cookie banner at multiple places in your dom tree if you wanted so I was generating random IDs to avoid dom collisions. This didn't make sense to allow for multiple cookie scripts on a single page so I removed the context provider aspects and moved this to a global const. I thought about removing the random bits but I left it in there since I pulled this component out into essex themed components to be reused across apps just in case the host app is for some reason using the ID "cookie-consent-banner". I am going to look at removing the need to use IDs and keep this detail more internal.

Reply:
I did an experiment to keep track/handle of the dom node useing useState inside the useEffect. This kind of works but causes unnecessary rerenders and issues since the dom nodes exist outside of the react dom and not using the React useRef way of keeping track of the elements. I think using unique IDs is the best approach since the elements live outside of react but I don't think we need to worry about how we are generating the random bits. I would like to keep it simple with Math.random so I don't have to include a dependency on lodash when it is a reusable component.

Reply:
Awesome, this makes sense Derek.